### ISSUE TITLE
[Accessibility] Screen Reader is not reading ShimmeredDetailsList's column header with associated cell value

### ISSUE THREAD

**Issue by kejatura** [OfficeDev/office-ui-fabric-react#11348]
### Environment Information

- **Package version(s)**: 7.31.2
- **Browser and OS versions**: [Edge, Chrome, FireFox] + Narrator and NVDA

### Describe the issue:

When using ShimmeredDetailsList component from office-ui-fabric-react, notice when an item is selected (using single selection mode), screen reader's scan mode is not reading the correct header information associated with cell value. This was addressed previously: 

https://github.com/OfficeDev/office-ui-fabric-react/issues/5164#issuecomment-409370385

### Please provide a reproduction of the issue in a codepen:

(copied from https://developer.microsoft.com/en-us/fabric#/controls/web/detailslist/shimmer)

https://codepen.io/kejatura/pen/GRgRxQe?editors=1010

- click toggle, and select an item in the list.
- Turn on screen reader's scan mode for selected item and read through the selection.

#### Actual behavior:

- User will notice that the column header reading is ahead of the current cell value.

#### Expected behavior:

- Column header should be read with associated cell value.
